README.md

```diff
@@ -75,69 +75,67 @@ And the following methods to support registration and receiving notifications:
+
+3. Add the library to your `MainApplication.java`:
 
 ```java
 import com.wix.reactnativenotifications.RNNotificationsPackage;
 
-...
+// ...
 
     @Override
     protected List<ReactPackage> getPackages() {
         return Arrays.<ReactPackage>asList(
             new MainReactPackage(),
-	        // ...
-	        new RNNotificationsPackage(MainApplication.this)
+            // ...
+            new RNNotificationsPackage(MainApplication.this)
         );
 ```
 
 ### Receiving push notifications
 
 > This section is only necessary in case you wish to **receive** push notifications in your React-Native app.
 
-Push notifications on Android are managed and dispatched using [Google's GCM service](https://developers.google.com/cloud-messaging/gcm) (now integrated into Firebase). The following installation steps are a TL;DR of [Google's GCM setup guide](https://developers.google.com/cloud-messaging/android/client). You can follow them to get GCM integrated quickly, but we recommend that you will in the very least have a peek at the guide's overview.
-
-#### Step #1: Subscribe to Google's GCM
+Push notifications on Android are managed and dispatched using Google's [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/) service.
 
-To set GCM in your app, you must first create a Google API-project and obtain a **Sender ID** and a **Server API Key**. If you have no existing API project yet, the easiest way to go about in creating one is using [this step-by-step installation process](https://developers.google.com/mobile/add); Use [this tutorial](https://code.tutsplus.com/tutorials/how-to-get-started-with-push-notifications-on-android--cms-25870) for insturctions.
+1. Add Firebase Messaging to your Android app in the [Firebase console](https://console.firebase.google.com/), Settings -> Cloud Messaging.
 
-Alternatively, follow [Google's complete guide](https://developers.google.com/cloud-messaging/android/client#create-an-api-project).
+2. [Integrate Firebase](https://firebase.google.com/docs/android/setup) in your native Android project.
 
-#### Step #2: Add Sender ID to Manifest File
-
-Once obtained, bundle the Sender ID onto your main `manifest.xml` file:
+If you've followed the instructions correctly your Android project will now include a file called `google-services.json` and the bottom of your `build.gradle` will look something like:
 
 ```gradle
-<manifest>
-...
-	<application>
-	...
-		// Replace '1234567890' with your sender ID.
-		// IMPORTANT: Leave the trailing \0 intact!!!
-	    <meta-data android:name="com.wix.reactnativenotifications.gcmSenderId" android:value="1234567890\0"/>
-	</application>
-</manifest>
 
+// ...
+
+apply plugin: 'com.google.gms.google-services'
 ```
 
+**IMPORTANT**: Do NOT add `com.google.firebase:firebase-messaging` as an explicit dependency unless you wish to try override our version (see "Installation").
```

Comment:
Not sure I understand this? In Installation it indicates to add as a dependency but then you're saying _not_ to here?

Reply:
@tommeier Nyah, crud. I edited the README several times. Originally I didn't want you to have to manually add a dependency. But the bug in Google's plugin meant it was impossible. I need to fix the README.

You *do need* a dependency.